src/compatibility.ts

```diff
@@ -74,3 +75,24 @@ export const requiredHubFlagWithDeprecations = requiredHubFlag({
   aliases: ['targetdevhubusername'],
   deprecateAliases: true,
 });
+
+/**
+ * @deprecated
+ */
+export const arrayWithDeprecation = (options: Record<string, unknown> = {}) =>
+  Flags.string({
+    // populate passed options
+    ...options,
+    // overlay those options we want to own
+    deprecated: { message: messages.getMessage('arrayStyleFlagsDeprecated') },
+    multiple: true,
+    parse: async (input: string) => {
+      if (input.includes(',')) {
+        const warningMsg = `${StandardColors.warning(messages.getMessage('warning.prefix'))} ${messages.getMessage(
+          'warning.arrayInputFormat'
+        )}`;
+        CliUx.ux.log(warningMsg);
```

Comment:
It's gonna fail really really early in the process, but won't using CliUx directly break json output if the user said `--json` ?

You could use Lifecycle.emitWarning so that "how should I display the warning" delegates to SfCommand, which knows what to do.